nativelib/src/main/resources/scala-native/libunwind/include-libunwind/__libunwind_config.h

```diff
@@ -10,99 +10,102 @@
+#define _LIBUNWIND_HIGHEST_DWARF_REGISTER_ARM 287
+#define _LIBUNWIND_HIGHEST_DWARF_REGISTER_OR1K 31
+#define _LIBUNWIND_HIGHEST_DWARF_REGISTER_MIPS 65
 
 #if defined(_LIBUNWIND_IS_NATIVE_ONLY)
-# if defined(__i386__)
-#  define _LIBUNWIND_TARGET_I386
-#  define _LIBUNWIND_CONTEXT_SIZE 8
-#  define _LIBUNWIND_CURSOR_SIZE 15
-#  define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_X86
-# elif defined(__x86_64__)
-#  define _LIBUNWIND_TARGET_X86_64 1
-#  if defined(_WIN64)
-#    define _LIBUNWIND_CONTEXT_SIZE 54
-#    define _LIBUNWIND_CURSOR_SIZE 66
-#  else
-#    define _LIBUNWIND_CONTEXT_SIZE 21
-#    define _LIBUNWIND_CURSOR_SIZE 33
-#  endif
-#  define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_X86_64
-# elif defined(__powerpc64__)
-#  define _LIBUNWIND_TARGET_PPC64 1
-#  define _LIBUNWIND_CONTEXT_SIZE 167
-#  define _LIBUNWIND_CURSOR_SIZE 179
-#  define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_PPC64
-# elif defined(__ppc__)
-#  define _LIBUNWIND_TARGET_PPC 1
-#  define _LIBUNWIND_CONTEXT_SIZE 117
-#  define _LIBUNWIND_CURSOR_SIZE 124
-#  define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_PPC
-# elif defined(__aarch64__)
-#  define _LIBUNWIND_TARGET_AARCH64 1
-#  define _LIBUNWIND_CONTEXT_SIZE 66
-#  define _LIBUNWIND_CURSOR_SIZE 78
-#  define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_ARM64
-# elif defined(__arm__)
-#  define _LIBUNWIND_TARGET_ARM 1
-#  if defined(__ARM_WMMX)
-#    define _LIBUNWIND_CONTEXT_SIZE 61
-#    define _LIBUNWIND_CURSOR_SIZE 68
-#  else
-#    define _LIBUNWIND_CONTEXT_SIZE 42
-#    define _LIBUNWIND_CURSOR_SIZE 49
-#  endif
-#  define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_ARM
-# elif defined(__or1k__)
-#  define _LIBUNWIND_TARGET_OR1K 1
-#  define _LIBUNWIND_CONTEXT_SIZE 16
-#  define _LIBUNWIND_CURSOR_SIZE 24
-#  define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_OR1K
-# elif defined(__mips__)
-#  if defined(_ABIO32) && _MIPS_SIM == _ABIO32 && defined(__mips_soft_float)
-#    define _LIBUNWIND_TARGET_MIPS_O32 1
-#    define _LIBUNWIND_CONTEXT_SIZE 18
-#    define _LIBUNWIND_CURSOR_SIZE 24
-#  elif defined(_ABIN32) && _MIPS_SIM == _ABIN32 && defined(__mips_soft_float)
-#    define _LIBUNWIND_TARGET_MIPS_NEWABI 1
-#    define _LIBUNWIND_CONTEXT_SIZE 35
-#    define _LIBUNWIND_CURSOR_SIZE 42
-#  elif defined(_ABI64) && _MIPS_SIM == _ABI64 && defined(__mips_soft_float)
-#    define _LIBUNWIND_TARGET_MIPS_NEWABI 1
-#    define _LIBUNWIND_CONTEXT_SIZE 35
-#    define _LIBUNWIND_CURSOR_SIZE 47
-#  else
-#    error "Unsupported MIPS ABI and/or environment"
-#  endif
-#  define _LIBUNWIND_HIGHEST_DWARF_REGISTER _LIBUNWIND_HIGHEST_DWARF_REGISTER_MIPS
-# else
-#  error "Unsupported architecture."
-# endif
+#if defined(__i386__)
+#define _LIBUNWIND_TARGET_I386
+#define _LIBUNWIND_CONTEXT_SIZE 8
+#define _LIBUNWIND_CURSOR_SIZE 15
```

Comment:
Is there a rule for clang-format to allow indented definitions that were existing before? Not a big deal if there is no such thing, but it better as easier to understand when it was structured

Reply:
I know this can be done, will find the option.

Reply:
Refer to https://clang.llvm.org/docs/ClangFormatStyleOptions.html
Search for `IndentPPDirectives `
I sort of like the last option better than the second: `PPDIS_BeforeHash`

What do you think?

Reply:
Gosh, after poking around on the Mac, most are just flush. I did see one file with the following `clapack.h` but it is not 4 spaces.
```
#include <stdint.h>
#if __has_include(<os/availability.h>)
# include <os/availability.h>
#else // __has_include(<os/availability.h>)
# undef API_AVAILABLE
# define API_AVAILABLE(...) /* Nothing */
#endif // __has_include(<os/availability.h>)
```
This doesn't seem to common so I'm wondering if we should just go with the defaults.

Reply:
I think we can go with the defaults with this one. 

The only problem is clang-format 6.0 does not support BeforeHash indent https://releases.llvm.org/6.0.0/tools/clang/docs/ClangFormatStyleOptions.html

Reply:
I think that is fine for now. It was late last night and I was looking at the latest.